src/Analyser/TypeSpecifier.php

```diff
@@ -924,8 +926,19 @@ private function specifyTypesForConstantBinaryExpression(
 				}
 				$argType = $scope->getType($exprNode->getArgs()[0]->value);
 				if ($argType->isArray()->yes()) {
+					if (count($exprNode->getArgs()) === 1) {
+						$isNormalCount = true;
+					} else {
+						$mode = $scope->getType($exprNode->getArgs()[1]->value);
+						$isNormalCount = (new ConstantIntegerType(COUNT_NORMAL))->isSuperTypeOf($mode)->yes();
+
+						if (!$isNormalCount) {
+							$isNormalCount = $argType->getIterableValueType()->isArray()->no();
```

Comment:
I think this still misses some scenarios.

We should have different results for `$isNormalCount` yes/maybe/no.
And also different results for `$argType->getIterableValueType()->isArray()` yes/maybe/no.

If you create a matrix:

no + no
no + maybe
no + yes
maybe + no
maybe + maybe
maybe + yes
yes + no
yes + maybe
yes + yes

You'll see that some scenarios aren't covered correctly.

Reply:
thanks, I added tests covering this testmatrix in 

`tests/PHPStan/Analyser/data/count-maybe.php`